drivers/scheduler/k8s/k8s.go

```diff
@@ -2929,32 +2966,104 @@ func (k *K8s) destroyBackupObjects(
 	return nil
 }
 
-// getExpectedAutopilotSizeForPvc calculates exected size of PVC based on autopilot rules and workload
-func (k *K8s) getExpectedAutopilotSizeForPvc(pvc *v1.PersistentVolumeClaim, apRules *apapi.AutopilotRuleList, wSize uint64) (uint64, error) {
-	var (
-		err             error
-		expectedPVCSize uint64
-	)
+// EstimatePVCExpansion calculates expected size of PVC based on autopilot rule and workload
+func (k *K8s) EstimatePVCExpansion(pvc *v1.PersistentVolumeClaim, apRule apapi.AutopilotRule, wSize uint64) (uint64, int, error) {
+	pvcObjSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+	pvcSize, _ := pvcObjSize.AsInt64()
+	expectedPVCSize := uint64(pvcSize)
 
-	driver, err := volume.Get(k.VolDriverName)
+	volDriver, err := volume.Get(k.VolDriverName)
 	if err != nil {
-		return 0, err
+		return 0, 0, err
+	}
+	if isAutopilotMatchPvcLabels(apRule, pvc) {
+		return volDriver.EstimateVolumeExpand(apRule, expectedPVCSize, wSize)
 	}
+	return expectedPVCSize, 0, nil
+}
 
-	pvcObjSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
-	pvcSize, _ := pvcObjSize.AsInt64()
-	expectedPVCSize = uint64(pvcSize)
+// ValidateAutopilotPVCEvents verifies proper alerts and events on resize completion
+func (k *K8s) ValidateAutopilotPVCEvents(ctx *scheduler.Context, coolDownPeriod int) error {
```

Comment:
instead of a fixed `coolDownPeriod`, we can take in a timeout argument. The implementation will retry until the timeout. By doing so, we can finish the validation earlier than the coolDownPeriod if we find the events.

Reply:
I need this argument because autopilot posts `ActiveActionsTaken => Normal` only after the cool down period, never earlier